ocaml/typing/env.ml

```diff
@@ -749,6 +772,16 @@ let error err = raise (Error err)
 let lookup_error loc env err =
   error (Lookup_error(loc, env, err))
 
+type actual_mode = {
```

Comment:
No longer used

Reply:
I still find it useful to have this type internal to `env.ml`, even just to reduce the number of arguments.

Reply:
Ok, if you really want to keep this type then probably the thing to do is revert the change I requested so it's still in the interface. Then rename it something like `identifier_mode` so it has a name with a clearer semantic meaning and then add versions of `submode` and `mode_cross_left` that operate directly on it.

That will fix my criticism of the earlier code. Previously the pair was only ever immediately deconstructed by the callers of `lookup_value`, which implies that it wasn't being used as an actual abstraction. If they instead start to treat it like a real value then it is being used as an actual abstraction.

Reply:
Yes I also thought about adding new version of `submode` and `mode_cross_left`, but felt they will be overhauled soon with a future refactoring of mode errors. So I didn't do that which left the abstraction not fully utilized.

I still find the simple packing useful - that being said, I will remove the packing to get the PR merged soon.

Reply:
Put it in another way: we already have many functions taking and returning this `actual_mode` - is that sort of abstraction? Then, I think partial abstraction is better than no abstraction.

Reply:
By "two lines long" I guess you mean to define them as wrappers of the existing functions in `typecore.ml`, which I have done now in the commit "more abstraction for actual_mode".

In a future PR I will move `actual_mode`, `expected_mode` and related functions to a single module that sits between `env/typecore` and `mode.ml`.